### ISSUE TITLE
Planning for an alpha release

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#376]
DeepChem is currently trending towards an alpha release. The goal of an alpha-release is to achieve basic API stability, so that we can start building up educational material and tutorials that explain how to use DeepChem in practice. Here are some of the things I'd like to see in an alpha release

- **A unified API for constructing models in Tensorflow:** It should be as easy to construct models with DeepChem as it is with Keras. Furthermore, these constructions should interoperate nicely with raw-tensorflow so sophisticated users can swap back-and-forth as desired. #369 makes partial progress towards this goal. Will need to be completed and merged.
- **A good story for protein/ligand preparation:** Preparing proteins and ligands for processing will require proper handling of water, ions, binding pocket pKa (or pocket protonation generally), ligand tautomers, and ligand resonance  structures. See #368 for story.
- **Full integration of one-shot models:** Should be feasible to save and robustly run one-shot models (#349, #292, #288). This should be handled by the API refactor, but breaking out as separate point to emphasize.
- **Replace OpenBabel with RDKit:** OpenBabel has some issues with segfaults (#329, #331, #338). Would be good to replace with RDKit. See #371 for details.
- **Fix broken IPython notebooks and add more:** DeepChem used to have tutorial ipython notebooks, but these were broken by the sequence of rapid refactorings (see `deepchem/examples/broken`). I think it would be worth our effort to get these fixed and set-up as a beginner resource for the alpha-relase. (#214)
- **Tensorflow 1.0 upgrade:** DeepChem is now pinned to Tensorflow 0.12. Should upgrade to 1.0 for alpha release (#398)

If there are any other asks for an alpha-release, please chime in! Tentative date for release is early March.

**Comment by proteneer** [comment]
I'd also push for cleaning up the transformers class a little bit. Both the constructor  and the transform() methods take in datasets - which is a bit confusing (and can lead to inconsistent behavior when people pass in different datasets by accident, eg. NormalizationTransformer has means computed from  the constructor, but the actual transform() can take in a different dataset.)

**Comment by rbharath** [comment]
The NormalizationTransformer is that way by design (the transform is
defined with respect to train set since that's only available data when
models are constructed, and is evaluated on test using mean/deviation from
train). It does sound though like we can put in better docs for
Transformers explaining why different datasets are allowable.

On Feb 2, 2017 12:45 PM, "Yutong Zhao" <notifications@github.com> wrote:

I'd also push for cleaning up the transformers class a little bit. Both the
constructor and the transform() methods take in datasets - which is a bit
confusing (and can lead to inconsistent behavior when people pass in
different datasets by accident, eg. NormalizationTransformer has means
computed from the constructor, but the actual transform() can take in a
different dataset.)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/deepchem/deepchem/issues/376#issuecomment-277077913>,
or mute the thread

**Comment by proteneer** [comment]
Ah I see - that makes *much* more sense now. In addition to the documentation, for the classes like the NormalizationTransform, maybe we can add an \__init\__() that mandates that the dataset arg is not None (it seems non-sensical to call this class on a non-existent dataset). (Of course in this \__init\__ it would still at some point call the base classes constructor through super())

**Comment by proteneer** [comment]
do we want to get the tensorflow 1.0.x patch into this as well?

**Comment by rbharath** [comment]
@proteneer I'd really like to, but this one might be out of our hands since it depends on when Stanford cluster admins update our Tensorflow. They said ETA 1 Month, but these are usually soft targets...

**Comment by proteneer** [comment]
hm - you can't hot swap or override out the tensorflow-gpu package from your venv's site-package? (it's what i've been doing)

**Comment by rbharath** [comment]
Unfortunately not. Our clusters have an old glibc, so basically impossible to install Tensorflow in user-space. The upgraded cluster (coming next month) should solve this issue for us. (Fingers crossed...)

**Comment by rbharath** [comment]
Added on #214 as a target for the alpha-release. Updated comment above.

**Comment by rbharath** [comment]
@proteneer Some of the Stanford clusters have upgraded, so added on Tensorflow 1.0 as a target for the alpha release. PRs welcome :-D

**Comment by lilleswing** [comment]
I think we can close this as we already have a 1.0.0 and a 1.1.0 release.

**Comment by rbharath** [comment]
Agreed!
